src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -1943,8 +1943,46 @@ void c_typecheck_baset::typecheck_side_effect_function_call(
     if(symbol_table.symbols.find(identifier)==symbol_table.symbols.end())
     {
       // This is an undeclared function.
+
+      // Is it the polymorphic assigns clause function ?
+      if(identifier == CPROVER_PREFIX "typed_target")
+      {
+        auto arg0 = expr.arguments().at(0);
+        typecheck_expr(arg0);
```

Comment:
Check that there is exactly one argument and else report back via `error()` and `throw`. It should not be possible for a user to trigger a system exception (`.at(0)`).